src/tactic/reassoc_axiom.lean

```diff
@@ -118,4 +127,21 @@ do n ← ident,
    do n ← resolve_constant n,
       reassoc_axiom n
 
+namespace interactive
+
+setup_tactic_parser
+
+/-- `reassoc h`, for assumption `h : x ≫ y = z ≫ x`, creates a new assumption `h : ∀ {W} (f : Z ⟶ W), x ≫ y ≫ f = z ≫ x ≫ f`.
```

Comment:
```suggestion
/-- `reassoc h`, for assumption `h : x ≫ y = z`, creates a new assumption `h : ∀ {W} (f : Z ⟶ W), x ≫ y ≫ f = z ≫ f`.
```
I don't think it's helpful to have `x` repeated, and if I understand correctly it's the form of the LHS that we're matching on, anyway.

Reply:
That's true. At the same time, having composition on both sides of `=` emphasizes that rebracketting happens on both sides.

Reply:
Oh, I hadn't actually realised that. Perhaps we better improve the specification/examples, then.

Reply:
I think the syntax I chose is also not great. Maybe writing `reassoc h` / `reassoc h' := h` instead of `reassoc! h` / `reassoc` would be better. Or maybe integrating it into a version of `simp` would be better still.